test/app_test_case.py

```diff
@@ -109,7 +110,7 @@ def setUp(self):
                 break
 
     def _ping(self):
-        return requests.get(f"{self.base_url}/health/basic")
+        return requests.get(self.base_url.set(path='/health/basic').url)
```

Comment:
There are still a million places that you touched where `.url` is used. So annoying!

Reply:
As per our PL discussion, we only addressed variables of `furl` like `url` that made use of the `.url` property, furthermore …
>I recall that we agreed to try `str(url)` instead of `url.url`.

seemed to me as a literal directive to address `url.url` -> `str(url)` only.

While I appreciate your feedback and I am profoundly grateful for your mentorship, I think `So annoying!` was unnecessary.

Reply:
> As per our PL discussion, we only addressed variables of `furl` like `url` that made use of the `.url` property, furthermore …
> 
> > I recall that we agreed to try `str(url)` instead of `url.url`.
> 
> seemed to me as a literal directive to address `url.url` -> `str(url)` only.

That's not what I meant and, to be honest, a naíve misinterpretation. There is no point in changing only literal occurrences of `url.url` to `str(url)`. The only sensible interpretation of my code fragment is to replace any use of the `url` property on any instance of the `furl` class to instead calling the `str` built-in on that instance. Anything else would create a completely meaningless and inconsistent mix of approaches.

> While I appreciate your feedback and I am profoundly grateful for your mentorship, I think `So annoying!` was unnecessary.

You, Trevor and I will discuss this offline.